src/rdf4cpp/rdf/Literal.hpp

```diff
@@ -334,6 +336,85 @@ class Literal : public Node {
     Literal to_node_storage(NodeStorage &node_storage) const noexcept;
     [[nodiscard]] Literal try_get_in_node_storage(NodeStorage const &node_storage) const noexcept;
 
+private:
+    [[nodiscard]] static NodeID find_datatype_iri(datatypes::registry::DatatypeIDView id, NodeStorage &node_storage) noexcept;
+
+public:
+    /**
+     * searches for a xsd::String Literal in the specified node storage and returns it.
+     * returns a null Literal, if not found.
+     * @param lexical_form
+     * @param node_storage
+     * @return
+     */
+    [[nodiscard]] static Literal find_simple(std::string_view lexical_form, NodeStorage &node_storage = NodeStorage::default_instance()) noexcept;
+
+    /**
+     * searches for a rdf::LangString Literal in the specified node storage and returns it.
+     * returns a null Literal, if not found.
+     * @param lexical_form
+     * @param node_storage
+     * @return
+     */
+    [[nodiscard]] static Literal find_lang_tagged(std::string_view lexical_form, std::string_view lang_tag, NodeStorage &node_storage = NodeStorage::default_instance()) noexcept;
+
+    /**
+     * searches for a Literal of type T in the specified node storage and returns it.
+     * returns a null Literal, if not found.
+     * if T is inlineable (and not rdf::LangString) always returns the inlined Literal.
+     * @tparam T
+     * @param compatible_value
+     * @param node_storage
+     * @return
+     */
+    template<datatypes::LiteralDatatype T>
+    [[nodiscard]] static Literal find(typename T::cpp_type const &compatible_value,
```

Comment:
Missing templated version that takes `string_view` (== `find_typed`)
+ rename this to `find_typed_from_value`